docs/lesson-plans/week-1.md

```diff
@@ -0,0 +1,726 @@
+
+Example:
+```
+# Move the README.md file to the parent directory without renaming it
+mv README.md ../
+
+# Rename the test.txt file to test.md
+mv test.txt test.md
+
+# Move the test.txt file to the Desktop from Documents and rename it to hello.world
+mv Documents/test.txt Desktop/hello.world
+```
+
+#### cp
+
+`cp` copies a file, and it's very similar to `mv`. 
+
+Format:
+```
+cp <exiting file or directory> <new file or directory>
+```
+
+- If you're copying a directory, you need to make sure to include the `-r` flag so that it 
+copies everything in the directory too.
+
+Examples:
+```
+# Copy test.txt to Desktop
+cp test.txt Desktop
+
+# Copy the Desktop directory to Documents 
+cp Desktop Documents -r
+```
+
+#### mkdir
+
+`mkdir` creates a directory.
+
+Format:
+```
+mkdir <directory name>
+```
+
+Examples:
+```
+# Create a Desktop2 directory
+mkdir Desktop2
+```
+
+#### rm
+
+`rm` deletes a file or a directory.
+
+```
+rm <path to file or directory>
+```
+
+- Be careful using this command. Once it's used, files are pretty much unrecoverable without special software.
+- If you're deleting a directory, you'll need to pass in the `-rf` flags.
+    - The `-r` flag makes it recursive, it will delete all subdirectories.
+    - The `-f` flag is called force, and it's sometimes required to delete a non-empty directory.
+- Never try to delete the `/` directory using `rm -rf /`. It is a very dangerous command that will
+end up deleting your entire computer (there are a couple protections, but would you really want
+to test them out?).
+
+```
+# Delete the test.txt file
+rm test.txt
+
+# Delete the test directory
+rm -rf test
+```
+
+#### cat
+
+`cat` prints out the contents of a file. It technically stands for 'concatenate' file to 
+standard output (your console).
+
+Format:
+```
+cat <filepath>
+```
+
+```
+# Print out contents of test.txt
+cat test.txt
+```
+
+#### vim/vi
+
+`vim` (sometimes also `vi`) is a text editor that usually comes preinstalled on most terminals. It allows you
+to open files, edit them using quick key binds, and write/close the files. It's notorious
+for having a really steep learning curve, so people tend to avoid it, but it can be a 
+powerful tool if you do end up spending time trying to learn it.
+
+![text-editor-learning-curves](./img/text-editor-learning-curves.png)
+
+We're showing this since you'll have to at some point learn a terminal-based text editor.
+It's also faster sometimes for quick edits to a file.
+
+Before you even try to open it, though, you need to know how to exit it. The easiest way to
+do that is to type `<esc>`, then `:q!`.
+
+Format:
+```
+# Open a file using vim
+vim <filepath>
+
+# vimtutor (see below)
+vimtutor
+```
+
+- You start in 'normal' mode
+- To edit a file, type 'i' to go into 'insert' mode.
+- To get back into normal mode, type <esc>
+- To save a file, type ":w" in normal mode
+- (again) To exit, type ":q!" in normal mode
```

Comment:
Doesn't vim give you a warning if you try `:q` without saving and the prompt includes saying the `:q!` prompt? If that's the case let's just tell them to use the q without the ! and I'm sure they'll figure it out

Reply:
Lmao ok.